fs/fs_zenfs.cc

```diff
@@ -1383,9 +1383,10 @@ IOStatus ZenFS::MigrateFileExtents(
   IOStatus s = IOStatus::OK();
   Info(logger_, "MigrateFileExtents, fname: %s, extent count: %lu",
        fname.data(), migrate_exts.size());
+
   // The file may be deleted by other threads, better double check.
   auto zfile = GetFile(fname);
-  if (zfile == nullptr || zfile->IsOpenForWR()) {
+  if (zfile == nullptr || zfile->IsDeleted() || zfile->IsOpenForWR()) {
```

Comment:
Maybe add a comment that the `IsDeleted` check is an optimistic optimization. Code is also valid without the check, but the check can help avoid redundant work.

Reply:
Will do, i'll rework the commit message as well to clarify the isdeleted usage

Reply:
I think this was dropped, so resolving comment.